gnucash/register/register-gnome/gnucash-register.c

```diff
@@ -161,6 +161,37 @@ gnucash_register_refresh_from_prefs (GnucashRegister *reg)
     gnc_header_request_redraw (GNC_HEADER(sheet->header_item));
 }
 
+void
+gnucash_register_reset_sheet_layout (GnucashRegister *reg)
+{
+    GNCHeaderWidths widths;
+    Table *table;
+    GList *node;
+    gchar *key;
+    guint value;
+    GnucashSheet *sheet;
+    gint current_width;
+
+    g_return_if_fail (reg != NULL);
+
+    sheet = GNUCASH_SHEET(reg->sheet);
+
+    g_return_if_fail (sheet != NULL);
+    g_return_if_fail (GNUCASH_IS_SHEET (sheet));
+
+    current_width = sheet->window_width - 1;
+
+    widths = gnc_header_widths_new ();
+    gnucash_sheet_set_header_widths (sheet, widths);
```

Comment:
It does not drop the state as the register will be saved on close unless I have not noticed that a default register layout does not save.

Reply:
Hmm, good point. Though I think that if we are currently also saving default state for each register we didn't change we definitely should stop doing that if we want to introduce a customizable default state.
Consider the following scenario assuming state is always saved per register (default or not) and we have the option to redefine a default state as well:
- open a register
- close it again. Assuming we always save state on close this will save the current default state, though we didn't really ask for it.
- now open another register, tweak the columns and save the new layout as default state
- reopen the first register. This register will still be use the original default state as that's what we saved.

However as the user never explicitly manipulated the state of that first register, s/he would probably expect it to use the new defaults.

Had we not saved the default state as register specific state when closing that first register, it would have picked up the new default and use that on reopening.